service-python/main_consumer.py

```diff
@@ -26,7 +26,8 @@ def __init__(self, config):
         self.setup_queues()
```

Comment:
Still calling this if self.connection is None. If I see self.connection to be None in garbage collection it can be None here too.